src/olympia/files/utils.py

```diff
@@ -310,6 +311,11 @@ def apps(self):
                 continue
             if app.guid not in amo.APP_GUIDS or app.id in seen_apps:
                 continue
+            if (app not in amo.APP_USAGE_FIREFOXES_ONLY and
+                    switch_is_active('disallow-thunderbird-and-seamonkey')):
+                # Ignore non-firefoxes compatibility (APP_GUIDS still contain
+                # thunderbird and seamonkey at the moment).
+                continue
```

Comment:
This seems like it should go in the follow-up commit to #8344

Reply:
Yeah, I thought I wouldn't need it in #8344 then writing this PR I realized I would. So this is kinda the follow-up :)

Reply:
PR #8568 now includes this bit. I'll remove it from here.